src/neo/SmartContract/Native/TransactionState.cs

```diff
@@ -0,0 +1,25 @@
+using Neo.IO;
+using Neo.Network.P2P.Payloads;
+using Neo.VM;
+using Neo.VM.Types;
+
+namespace Neo.SmartContract.Native
+{
+    public class TransactionState : IInteroperable
+    {
+        public uint BlockIndex;
+        public Transaction Transaction;
```

Comment:
Missing `VMState`?

Reply:
Yes. I removed it.

Reply:
I think that it's useful for explorers know if the transaction was fault.

Reply:
Explorers can use `ApplicationLogs`.

Reply:
I think that only cost one byte and it's a useful information

Reply:
If you need `VMState`, in`LedgerContract.OnPersist` you have to run the script of all the transactions.

Reply:
How to get `transaction.BlockIndex`, Is this correct?
```c#
TransactionState txState = Blockchain.Singleton.View.Transactions.TryGet(hash);
var blockIndex = txState.BlockIndex;

↓

using var snapshot = Blockchain.Singleton.GetSnapshot();
Transaction tx = NativeContract.Ledger.GetTransaction(snapshot, hash);
TransactionState txState = new TransactionState()
{
    Transaction = tx,
    BlockIndex = NativeContract.Ledger.CurrentIndex(snapshot) - tx.ValidUntilBlock
};
var blockIndex = txState.BlockIndex;
```